plugins/resources/src/unixfilesystem.cpp

```diff
@@ -83,6 +87,198 @@
 const std::string DEFAULT_VAULT_DIR_MODE( "default_vault_directory_mode_kw" );
 const std::string HIGH_WATER_MARK( "high_water_mark" ); // no longer used
 const std::string REQUIRED_FREE_INODES_FOR_CREATE("required_free_inodes_for_create"); // no longer used
+const std::string HOST_MODE("HOST_MODE");
+const std::string HOST_LIST("HOST_LIST");
+
+namespace {
+    bool is_operating_in_detached_mode(irods::plugin_property_map& prop_map)
+    {
+
+        std::string host_mode_str;
+
+        irods::error ret = prop_map.get< std::string >(HOST_MODE, host_mode_str);
+        if (ret.ok()) {
+            if (boost::iequals(host_mode_str.c_str(), "detached"))  {
+                return true;
+            }
+        }
+
+        // default is attached mode
+        return false;
+    }
+}
+
+static std::vector<std::string> split(const std::string str, char delim)
```

Comment:
Is there a particular reason to define a static `split` function here rather than using the existing `irods::string_tokenize`?

If explicitly using a `char`-typed delimiter is desired, I would suggest overloading `irods::string_tokenize`.
If `irods::string_tokenize` is in need of modernization, then let's do that.

Reply:
I didn't know about it or didn't think of it.  That is much better than creating my own.  Thanks for pointing it out.  It was seamless to swap that out.